lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -3417,83 +3417,83 @@ class ConvertAtenContiguousOp : public OpConversionPattern<AtenContiguousOp> {
-
-    bool pinMemory = false;
-    if (!pin_memory.getType().isa<Torch::NoneType>() &&
-        !matchPattern(pin_memory, m_TorchConstantBool(&pinMemory))) {
-      return rewriter.notifyMatchFailure(
-          op, "pin_memory must be constant bool or None");
+    // Memory formats are not supported in the case of `AtenEmptyMemoryFormat`.
+    if constexpr (std::is_same<OpTy, AtenEmptyMemoryFormatOp>::value) {
+      if (!op.memory_format().getType().template isa<Torch::NoneType>())
+        return rewriter.notifyMatchFailure(
+            op, "unimplemented: only default memory format is supported");
     }
-    if (pinMemory)
-      return rewriter.notifyMatchFailure(op, "memory pinning not supported");
 
-    SmallVector<Value> sizes, sizeIndex;
-    if (!getListConstructElements(size, sizes)) {
+    Location loc = op.getLoc();
+    TypeConverter *typeConverter = this->getTypeConverter();
+    SmallVector<Value> resultSizeTorchInt, resultSize, resultSizeIndex;
+    if (!getListConstructElements(op.size(), resultSizeTorchInt)) {
       return rewriter.notifyMatchFailure(
-          op, "size must be created by ListConstruct");
+          op, "unimplemented: size must be constructed using ListConstruct");
     }
-    sizes = getTypeConvertedValues(rewriter, loc, getTypeConverter(), sizes);
-    for (size_t i = 0; i < sizes.size(); i++)
-      sizeIndex.push_back(castIntToIndex(rewriter, loc, sizes[i]));
+    resultSize = getTypeConvertedValues(rewriter, loc, typeConverter,
+                                        resultSizeTorchInt);
+    for (auto size : resultSize)
+      resultSizeIndex.push_back(castIntToIndex(rewriter, loc, size));
 
-    RankedTensorType newResultType =
-        getTypeConverter()
-            ->convertType(op->getResult(0).getType())
-            .cast<RankedTensorType>();
-    Type outElementType = newResultType.getElementType();
-
-    Value constantOp = rewriter.create<arith::ConstantOp>(
-        loc, outElementType,
-        (outElementType.isa<mlir::FloatType>()
-             ? rewriter.getFloatAttr(outElementType, elementValue)
-                   .cast<mlir::Attribute>()
-             : rewriter.getIntegerAttr(outElementType, elementValue)
-                   .cast<mlir::Attribute>()));
-    Value outTensor = rewriter
-                          .create<linalg::InitTensorOp>(
-                              loc, sizeIndex, newResultType.getElementType())
-                          .getResult();
-    Value fillOp = rewriter.create<linalg::FillOp>(loc, constantOp, outTensor)
-                       .getResult(0);
-
-    rewriter.replaceOpWithNewOp<tensor::CastOp>(op, newResultType, fillOp);
+    auto resultType =
+        typeConverter->convertType(op.getType()).template cast<RankedTensorType>();
+    Type outElemType = resultType.getElementType();
+
+    // Returns a constant of type `elemType` with value `val`.
+    auto getConstant = [&](int64_t val, Type elemType) -> Value {
```

Comment:
This is a very generic function and can share functionalities across many functions. Instead of a lambda can be placed outside as a static function.